lib/puppet/type/mysql_user.rb

```diff
@@ -46,6 +46,18 @@
   newproperty(:password_hash) do
     desc 'The password hash of the user. Use mysql_password() for creating such a hash.'
     newvalue(%r{\w*})
+
+    def change_to_s(currentvalue, _newvalue)
+      (currentvalue == :absent) ? 'created password' : 'changed password'
+    end
+
```

Comment:
Based on Robocop - [PredicateName](http://www.rubydoc.info/github/bbatsov/rubocop/Rubocop/Cop/Style/PredicateName) you cannot use `is_`, you can fix it by disable `Style/PredicateName` and enable it after `is_to_s`
`# rubocop:disable Style/PredicateName`

Reply:
This look's good. As a note for the future though you can put the disable on the line that violates the rule itself. This will make it only disable for that line. It's only if it is placed on an empty line that it need's to be reenabled.